src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestChangePolicyActionIT.kt

```diff
@@ -307,35 +307,39 @@ class RestChangePolicyActionIT : IndexStateManagementRestTestCase() {
         }
 
         // We should expect the explain API to show an initialized ManagedIndexMetaData with the default state from the initial policy
-        val explainResponseMap = getExplainMap(index)
-        assertPredicatesOnMetaData(
-            listOf(
-                index to listOf(
-                    ManagedIndexSettings.POLICY_ID.key to policy.id::equals,
-                    ManagedIndexMetaData.INDEX to executedManagedIndexConfig.index::equals,
-                    ManagedIndexMetaData.INDEX_UUID to executedManagedIndexConfig.indexUuid::equals,
-                    ManagedIndexMetaData.POLICY_ID to executedManagedIndexConfig.policyID::equals,
-                    StateMetaData.STATE to fun(stateMetaDataMap: Any?): Boolean =
-                        assertStateEquals(StateMetaData(policy.defaultState, Instant.now().toEpochMilli()), stateMetaDataMap)
-                )
-            ), explainResponseMap, false)
+        waitFor {
+            val explainResponseMap = getExplainMap(index)
+            assertPredicatesOnMetaData(
+                listOf(
+                    index to listOf(
+                        ManagedIndexSettings.POLICY_ID.key to policy.id::equals,
+                        ManagedIndexMetaData.INDEX to executedManagedIndexConfig.index::equals,
+                        ManagedIndexMetaData.INDEX_UUID to executedManagedIndexConfig.indexUuid::equals,
+                        ManagedIndexMetaData.POLICY_ID to executedManagedIndexConfig.policyID::equals,
+                        StateMetaData.STATE to fun(stateMetaDataMap: Any?): Boolean =
+                            assertStateEquals(StateMetaData(policy.defaultState, Instant.now().toEpochMilli()), stateMetaDataMap)
+                    )
+                ), explainResponseMap, false)
+        }
 
-        val changePolicy = ChangePolicy(newPolicy.id, null, emptyList(), false)
-        val response = client().makeRequest(RestRequest.Method.POST.toString(),
+        waitFor {
+            val changePolicy = ChangePolicy(newPolicy.id, null, emptyList(), false)
+            val response = client().makeRequest(RestRequest.Method.POST.toString(),
```

Comment:
I am wondering should we write this request inside waitFor block?

Reply:
This might be leftover from earlier testing before the other PR with ChangePolicy changes went in.

What was happening was that inside the ChangePolicy RestHandler it was doing a search and getting back an older document with a lower primaryTerm/seqNo which was causing the update to fail and the response coming back was failures: true, failed_indices: [<Something about mismatch between seqNo/PrimaryTerm>]. So this just ensured that the ChangePolicy was correctly applied by retrying if it got a failure because of that race condition. Will rerun a few times to see if thats fixed though.